scripts/records.js

```diff
@@ -0,0 +1,194 @@
+// 입력받은 정보를 토대로 삽입 / 수정 / 삭제 기능을 담당하는 함수들을 다루는 파일
+import { elements } from "./elements.js";
+
+export const records = [
+  {
+    date: "2025-08-14",
+    items: [
+      {
+        category: "문화/여가",
+        description: "스트리밍 서비스 정기 결제",
+        payment: "현대카드",
+        amount: -10900,
+      },
+      {
+        category: "교통",
+        description: "후불 교통비 결제",
+        payment: "현대카드",
+        amount: -45340,
+      },
+    ],
+  },
+  {
+    date: "2025-08-13",
+    items: [
+      {
+        category: "미분류",
+        description: "온라인 세미나 신청",
+        payment: "현대카드",
+        amount: -10000,
+      },
+    ],
+  },
+  {
+    date: "2025-08-10",
+    items: [
+      {
+        category: "식비",
+        description: "잔치국수와 김밥",
+        payment: "현대카드",
+        amount: -9500,
+      },
+      {
+        category: "월급",
+        description: "8월 급여",
+        payment: "현금",
+        amount: 2010580,
+      },
+    ],
+  },
+];
+
+// records 배열의 각 날짜별로 section을 만드는 함수
+export const renderRecords = (currentYear, currentMonth, records) => {
+  // 이전 렌더링 초기화
+  const recordContainerEl = elements.recordContainerEl();
+  recordContainerEl.innerHTML = "";
+
+  // 현재 헤더의 날짜에 해당하는 date값만 함수 호출
+  records.forEach((record) => {
+    const date = record.date;
+    //"YYYY-MM-DD" 에서 YYYY와 MM 추출 후 헤더의 날짜와 비교해서 같은 값만 호출
+    let year = date.split("-")[0];
+    let month = date.split("-")[1];
+    const items = record.items;
+    if (Number(currentYear) === Number(year) && Number(currentMonth) === Number(month)) {
+      renderRecordByDate({ date, items });
+    }
+  });
+};
+
+// 날짜와 데이터를 받아와서 섹션을 렌더링
+export const renderRecordByDate = ({ date, items }) => {
+  const formattedDate = getFormattedDate(date);
+  const recordsHTML = generateRecordHTML(items);
+
+  // 해당 날짜의 총 지출/수입을 구하기 위한 코드
+  const { income, outcome } = getTotalAmount(items);
+  let incomeContent = "";
+  let outcomeContent = "";
+  if (income !== 0) incomeContent = `수입 ${formatWithComma(income)}원`;
+  if (outcome !== 0) outcomeContent = `지출 ${formatWithComma(outcome)}원`;
+
+  const recordContainerEl = elements.recordContainerEl();
+  recordContainerEl.innerHTML += `
+    <div class="record-container">
+      <div class="record-header">
+        <div class="record-date">${formattedDate}</div>
+        <div class="record-amount">${incomeContent}  ${outcomeContent}</div>
+      </div> ${recordsHTML}
+    <div>
```

Comment:
닫힘 태그가 잘못 작성된 것 같아요!
```suggestion
    </div>
```

Reply:
필터링 부분 빠르게 임시로 만드느라 확인을 못했네요 ㅠ 확인 감사합니다 !